plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerFacetImpl.java

```diff
@@ -0,0 +1,669 @@
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+import org.sonatype.nexus.common.io.TempStreamSupplier;
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.InvalidContentException;
+import org.sonatype.nexus.repository.docker.DockerFacet;
+import org.sonatype.nexus.repository.http.HttpStatus;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.ComponentQuery;
+import org.sonatype.nexus.repository.storage.ComponentQuery.Builder;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.util.TypeTokens;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.ContentTypes;
+import org.sonatype.nexus.repository.view.Payload;
+import org.sonatype.nexus.repository.view.payloads.BlobPayload;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.nexus.transaction.Transactional;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import com.google.common.base.Supplier;
+import com.google.common.collect.Lists;
+import com.google.common.hash.Funnels;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+import com.google.common.io.ByteStreams;
+import com.orientechnologies.common.concur.ONeedRetryException;
+import com.orientechnologies.orient.core.storage.ORecordDuplicatedException;
+import org.apache.commons.lang.StringUtils;
+
+import static java.util.Collections.singletonList;
+import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA256;
+import static org.sonatype.nexus.repository.docker.internal.DockerAttributes.*;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_NAME;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_PATH;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_VERSION;
+
+/**
+ * {@link DockerFacet} implementation.
+ *
+ * @since 3.0
+ */
+@Named
+public class DockerFacetImpl
+    extends FacetSupport
+    implements DockerFacet
+{
+
+  private final static List<HashAlgorithm> hashAlgorithms = Lists.newArrayList(SHA256);
+
+  private final static ObjectMapper mapper = new ObjectMapper();
+
+  private final static Pattern validLayerId = Pattern.compile("^([a-f0-9]{16}|[a-f0-9]{64})$");
+
+  private final static Pattern validTag = Pattern.compile("^[\\w][\\w.-]{0,127}$");
+
+
+  @Override
+  @Transactional(retryOn = IllegalStateException.class)
+  public Payload getImagesIndex(final String name) throws IOException {
+    StorageTx tx = UnitOfWork.currentTransaction();
+
+    String imagesIndexAssetName = name + "/images";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/images" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)